File: AppTool/src/main/java/com/luoye/apptool/databinding/BaseDataAllFragment.java
package com.luoye.apptool.databinding;

import android.app.Activity;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.databinding.DataBindingUtil;
import androidx.databinding.ViewDataBinding;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.DefaultLifecycleObserver;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.luoye.apptool.utils.TClassUtils;

public abstract class BaseDataAllFragment<B extends ViewDataBinding, A_activity extends ViewModel, M_my extends ViewModel> extends Fragment implements DefaultLifecycleObserver {

    protected String TAG = "---BaseFragment";
    protected Context context;
    protected Activity activity;
    protected B binding;
    protected M_my viewModel;
    protected A_activity activityViewModel;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        context = getActivity();
        activity = getActivity();
        getLifecycle().addObserver(this);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, setLayoutId(), container, false);
        viewModel = new ViewModelProvider(this).get((Class<M_my>) TClassUtils.getTClass(getClass(), 2));
        activityViewModel = new ViewModelProvider(getActivity()).get((Class<A_activity>) TClassUtils.getTClass(getClass(), 1));
        binding.setLifecycleOwner(this);
        initFragment();
        return binding.getRoot();
    }


    protected abstract int setLayoutId();

    protected abstract void initFragment();

}

File: ktBase/src/main/java/com/luoye/ktbase/binding/fragment/databinding/BaseDataAllFragment.kt
package com.luoye.ktbase.binding.fragment.databinding

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.ViewDataBinding
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModel
import com.luoye.ktbase.binding.createActivityViewModel
import com.luoye.ktbase.binding.createViewModel
import com.luoye.ktbase.binding.getViewBinding

/**
 *
 * Created by: 
 * Time: 2023/4/7
 * user: 绑定自己ViewModel 绑定activity 共享ViewModel
 *
 */
abstract class BaseDataAllFragment<V : ViewDataBinding, MA : ViewModel,M : ViewModel> : Fragment() {


    protected lateinit var binding: V
        private set
    protected val viewModelActivity by lazy { createActivityViewModel<MA>(this, 0) }
    protected val viewModel by lazy { createViewModel<M>(this, 1) }


    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = getViewBinding(inflater, container, 0)
        binding.lifecycleOwner = this
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initFragment()
    }


    protected abstract fun initFragment()
}